src/filesystem.ts

```diff
@@ -105,6 +113,70 @@ export async function loadFileSystem({ config, dependencies, rootKey, username }
   return fs
 }
 
+/**
+ * Recover a user's file system.
+ */
+export async function recoverFileSystem({
+  agentDID,
+  auth,
+  dependencies,
+  oldUsername,
+  newUsername,
+  readKey,
+}: {
+  agentDID: () => Promise<string>
+  auth: AuthenticationStrategy
+  dependencies: Dependencies
+} & RecoverFileSystemParams): Promise<{ success: boolean }> {
+  try {
+    const { crypto, reference, storage } = dependencies
+
+    const newRootDID = await agentDID()
```

Comment:
I would call `DID.agent(dependencies.crypto)` from `import * as DID from "./did/index.js"` instead of passing `agentDID` as a parameter.